website/integrations: emby: add emby integration (#13774)
goauthentik/authentik: website/integrations/services/emby/index.md

1. If you don't have one already, create an LDAP bind user
2. Allow search access for the LDAP bind user on the application
    - Edit the provider
    - Select the tab **Permissions**
    - Select User **Object Permissions**
    - Select **Assign new user** and select you LDAP bind user with "Search full LDAP directory" and "Can view LDAP Provider" enabled
3. Add the LDAP bind user to the Application Bindings

## Emby configuration

1. Navigate to your Emby installation and log in with the admin account or currently configured local admin
2. Open the **Administrator dashboard** and go to the **Plugins** section
3. Click **Catalog** at the top of the page, and locate the "LDAP" plugin _(Needs Emby Premiere)_
4. Install the plugin. You may need to restart Emby to finish installation
5. Once finished, navigate back to the plugins section of the admin dashboard and click on the "LDAP" plugin icon to open the settings
6. Configure the LDAP Settings as follows:
    - `LDAP server address`: `ldap.company`
    - `LDAP server Port number`: `636`

>     - `Enable SSL`: **Checked**

Reviewer: ```suggestion
    - **Enable SSL**:  Toggled on
```

>     - `SSL certificate thumbprint (SHA1):`:

Reviewer: ```suggestion
    - **SSL certificate thumbprint (SHA1)**:
```

>         - Paste the SHA1 Fingerprint of your LDAP outpost Certificate here. It can be found on you Authentik Instance via **System** -> **Certificates**

Reviewer: Please include this on the previous line. And:
```suggestion
        - Enter the **SHA1 Fingerprint** of your LDAP outpost certificate in this field. You can obtain the fingerprint from your authentik instance by navigating to **System** > **Certificates** and expanding the certificate being used.
```

>     - `Bind DN`: `cn=ldap_bind_user,ou=users,dc=company,dc=com` _(Change to the LDAP bind user)_

Reviewer: ```suggestion
    - **Bind DN**: <kbd>cn=<em>ldap_bind_user</em>,ou=users,dc=<em>company</em>,dc=<em>com</em></kbd>
```

>     - `Bind credentials`: The Password of the user. If using a Service account, this is the token

Reviewer: ```suggestion
    - **Bind credentials**: The Password of the user. If using a Service account, this is the token
```

Also, it's best practice (and mabye required, I'd have to check) to use a (dedicated) service account. Could you reference documentation for service account creation and tell the user to input the token here. You should also tell the user how to get their SA token. You should be able to reference this doc directly for most of my suggestions: https://docs.goauthentik.io/docs/sys-mgmt/service-accounts

`ldap_bind_user`'s refernece in the preparation section should also be updated to reference SAs

>     - `User search base`: The base DN for LDAP queries. To query all users, set this to `dc=company,dc=com`

Reviewer: ```suggestion
    - **User search base**: The base Distinguished Name (DN) for LDAP queries. To query all users, set this to <kbd>dc=<em>company</em>,dc=<em>com</em></kbd>.
```

>         - You can specify an OU if you divide your users up into different OUs and only want to query a specific OU

Reviewer: ```suggestion
        - You can specify an Organizational Unit (OU) if you organize your users into different OUs and want to query only a specific one.
```

This *could* be added to the previous line, and could you also add an example like you did with the base DN example?

>     - `User search filter`: This is used to a user filter on what users are allowed to login. **This must be set**

Reviewer: I can't seem to be able to highlight this entire block for a suggestion, but I recommend:

```
    - **User Search Filter**: This setting is crucial for defining which users are allowed to log in.
         - **Allowing All Users**: Use the filter <kbd>(&(objectClass=user)(cn={0}))</kbd> to permit all users.
         - **Allowing Specific Group Members**: To restrict access to users in a particular group, use <kbd>(&(objectClass=user)(memberOf=cn=<em>emby_users</em>,ou=groups,dc=<em>company</em>,dc=<em>com</em>)(cn={0}))</kbd>.
```

        - To allow all users: `(&(objectClass=user)(cn={0})`
        - To only allow users in a specific group: `(&(objectClass=user)(memberOf=cn=emby_users,ou=groups,dc=company,dc=com)(cn={0}))` _(Change to the LDAP bind user)_
        - Good Docs on LDAP Filters: [atlassian.com](https://confluence.atlassian.com/kb/how-to-write-ldap-search-filters-792496933.html)

> 7. Click "Save"

Reviewer: ```suggestion
7. Click **Save**.
```

## Configuration verification

> Logout, and login with a LDAP user. Username **must** be used, logging in with email will not work

Reviewer: ```suggestion
Log out of Emby and attempt to log back in using your LDAP credentials from authentik. Ensure you use your **username** instead of your email address, as logging in with an email will not work.
```

What do you think?

> If there are issues with the login, you can try to debug your search filter. Good docs can be found here: [docs.goauthentik.io](https://docs.goauthentik.io/docs/add-secure-apps/providers/ldap/generic_setup#ldapsearch-test)

Reviewer: This could be moved to a dedicated "possible configuration errors" section, what do you think? Or mabye linking a common LDAP error page if we have that, I'm not sure
